core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -100,13 +106,23 @@ object ZkUtils {
     (zkClient, zkConnection)
   }
 
-  def DefaultAcls(isSecure: Boolean): java.util.List[ACL] = if (isSecure) {
-    val list = new java.util.ArrayList[ACL]
-    list.addAll(ZooDefs.Ids.CREATOR_ALL_ACL)
-    list.addAll(ZooDefs.Ids.READ_ACL_UNSAFE)
-    list
-  } else {
-    ZooDefs.Ids.OPEN_ACL_UNSAFE
+  def SensitivePath(path: String): Boolean = {
+    path == null || !SensitiveZkRootPaths.forall(!path.startsWith(_))
```

Comment:
No, null check was mainly to prevent NPE. I have changed to return false for null.